### PR TITLE
Switch from naive LRU implementation to restcache

### PR DESCRIPTION
**Pull request by ajacksified**
Prepare for using https://github.com/ajacksified/restcache in clients that use snoode, instead of implementing it here.

:eyeglasses: @curioussavage / @dwick, although not ready to merge until it's ready to implement in reddit-mobile as part of https://github.com/reddit/reddit-mobile/pull/301

Oh, and apparently I forked off of an earlier PR, so I'll have to fix that.

### CODE REVIEW

File: src/endpoints/v1.es6.js [hunk 1]
```diff
@@ -164,29 +145,12 @@ class APIv1Endpoint {
...
-      activity: new LRU(defaultCacheConfig),
-      saved: new LRU(defaultCacheConfig),
-      hidden: new LRU(defaultCacheConfig),
-      notifications: new LRU(defaultCacheConfig),
-      messages: new LRU(defaultCacheConfig),
-    };
   }
 
   hydrate (endpoint, options, data) {
-    var cache = this.cache[endpoint];
```
Comment (dwick) [inline src/endpoints/v1.es6.js]: is this method a noop now?

**Comment by dwick** [comment]
:haircut:

**Comment by dwick** [comment]
:nail_care:

**Comment by curioussavage** [comment]
:fish:

**Comment by dwick** [comment]
:fish:

**Pull request merged**